spec/services/declarations/create_spec.rb

```diff
@@ -0,0 +1,216 @@
+RSpec.describe Declarations::Create do
+  let(:author) { Events::LeadProviderAPIAuthor.new(lead_provider:) }
+  let(:lead_provider) { training_period.lead_provider }
+  let(:teacher) { training_period.trainee.teacher }
+  let!(:declaration_type) { "started" }
+  let(:evidence_type) { "training-event-attended" }
+  let(:schedule) { training_period.schedule }
+  let!(:milestone) { FactoryBot.create(:milestone, declaration_type:, schedule:) }
+  let(:declaration_datetime) { Faker::Time.between(from: milestone.start_date, to: milestone.milestone_date) }
+  let(:declaration_date) { declaration_datetime.iso8601 }
+  let(:contract_period) { training_period.contract_period }
+  let(:active_lead_provider) { training_period.active_lead_provider }
+  let(:delivery_partner) { training_period.delivery_partner }
+  let(:payment_statement) { FactoryBot.create(:statement, :open, active_lead_provider:) }
+  let(:existing_duplicate_declaration) { nil }
+
+  let(:service) do
+    described_class.new(
+      author:,
+      lead_provider:,
+      teacher:,
+      training_period:,
+      declaration_date:,
+      declaration_type:,
+      evidence_type:,
+      payment_statement:,
+      mentorship_period:,
+      delivery_partner:,
+      existing_duplicate_declaration:
+    )
+  end
+
+  describe "#create" do
+    subject(:create_declaration) { service.create }
+
+    %i[ect mentor].each do |trainee_type|
+      context "for #{trainee_type}" do
+        let(:teacher_type) { trainee_type }
+
+        context "when there's no existing declaration" do
+          let(:at_school_period) { FactoryBot.create(:"#{trainee_type}_at_school_period", started_on: 6.months.ago, finished_on: 2.weeks.from_now) }
+          let!(:training_period) { FactoryBot.create(:training_period, :"for_#{trainee_type}", :active, "#{trainee_type}_at_school_period": at_school_period, started_on: at_school_period.started_on, finished_on: at_school_period.finished_on) }
+          let!(:mentorship_period) do
+            if trainee_type == :ect
+              mentor = FactoryBot.create(:mentor_at_school_period, started_on: at_school_period.started_on, finished_on: at_school_period.finished_on)
+              FactoryBot.create(:mentorship_period, mentee: at_school_period, mentor:, started_on: at_school_period.started_on, finished_on: at_school_period.finished_on)
+            end
+          end
+
+          it "creates a new declaration with correct attributes" do
+            declaration = nil
+            expect { declaration = create_declaration }.to change(Declaration, :count).by(1)
+
+            expect(declaration.payment_statement).to be_nil
+            if trainee_type == :ect
+              expect(declaration.mentorship_period).to eq(mentorship_period)
+            end
+            expect(declaration.evidence_type).to eq(evidence_type)
+            expect(declaration.payment_status_eligible?).to be(false)
+          end
+
+          if trainee_type == :mentor
```

Comment:
FWIW the mentor completion service is safe to call for ECTs as well, so we could drop the check in the calling service and here